engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanParameter.java

```diff
@@ -63,10 +67,14 @@ public AnnotatedType getAnnotatedType() {
 	@Override
 	public Annotation[] getDeclaredAnnotations() {
 		try {
+			if ( annotationsForJDK8303112 != null ) {
+				// Working around https://bugs.openjdk.org/browse/JDK-8303112
+				return annotationsForJDK8303112.clone();
```

Comment:
We can remove the `.clone()` if you're absolutely certain nobody will ever try to mutate the array.